src/main/java/uk/gov/hmcts/reform/managecase/api/controller/NoticeOfChangeController.java

```diff
@@ -234,4 +252,72 @@ public RequestNoticeOfChangeResponse requestNoticeOfChange(@Valid @RequestBody R
         return noticeOfChangeService.requestNoticeOfChange(noCRequestDetails);
     }
 
+    @PostMapping(path = CHECK_NOTICE_OF_CHANGE_APPROVAL_PATH, produces = APPLICATION_JSON_VALUE)
+    @ApiOperation(value = "Check for Notice of Change Approval", notes = "Check for Notice of Change Approval")
+    @ApiResponses({
+        @ApiResponse(
+            code = 200,
+            message = StringUtils.EMPTY
+        ),
+        @ApiResponse(
+            code = 400,
+            message = "One or more of the following reasons:"
+                + "\n1) " + CASE_ID_INVALID
+                + "\n2) " + CASE_ID_INVALID_LENGTH
+                + "\n3) " + CASE_ID_EMPTY
+                + "\n4) " + CHANGE_ORG_REQUEST_FIELD_MISSING_OR_INVALID
+                + "\n5) " + NOC_DECISION_EVENT_UNIDENTIFIABLE,
+            response = ApiError.class,
+            examples = @Example({
+                @ExampleProperty(
+                    value = "{\n"
+                        + "   \"status\": \"BAD_REQUEST\",\n"
+                        + "   \"message\": \"" + CASE_ID_EMPTY + "\",\n"
+                        + "   \"errors\": [ ]\n"
+                        + "}",
+                    mediaType = APPLICATION_JSON_VALUE)
+            })
+        ),
+        @ApiResponse(
+            code = 401,
+            message = AuthError.AUTHENTICATION_TOKEN_INVALID
+        ),
+        @ApiResponse(
+            code = 403,
+            message = AuthError.UNAUTHORISED_S2S_SERVICE
+        )
+    })
+    public ResponseEntity checkNoticeOfChangeApproval(@Valid @RequestBody CheckNoticeOfChangeApprovalRequest
+                                                              checkNoticeOfChangeApprovalRequest) {
+        CaseDetails caseDetails = checkNoticeOfChangeApprovalRequest.getCaseDetails();
+        Optional<JsonNode> changeOrganisationRequestFieldJson = caseDetails.findChangeOrganisationRequestNode();
+
+        if (changeOrganisationRequestFieldJson.isEmpty()) {
+            throw new ValidationException(CHANGE_ORG_REQUEST_FIELD_MISSING_OR_INVALID);
+        }
+
+        ChangeOrganisationRequest changeOrganisationRequest =
+            jacksonUtils.convertValue(changeOrganisationRequestFieldJson.get(), ChangeOrganisationRequest.class);
+
+        validateNoCApproval(changeOrganisationRequest);
+
+        if (!changeOrganisationRequest.getApprovalStatus().equals(APPROVED_NUMERIC)
+            && !changeOrganisationRequest.getApprovalStatus().equals(APPROVED)) {
+            return ResponseEntity.ok().build();
```

Comment:
200 in this case?

Reply:
Yeah I checked with Matt and he said that we quietly return OK because it isn't an error